flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/runner/DefaultDispatcherRunnerITCase.java

```diff
@@ -173,8 +175,7 @@ private DispatcherGateway electLeaderAndRetrieveGateway(UUID firstLeaderSessionI
      * fail.
      */
     @Test
-    public void leaderChange_withBlockingJobManagerTermination_doesNotAffectNewLeader()
-            throws Exception {
+    void leaderChange_withBlockingJobManagerTermination_doesNotAffectNewLeader() throws Exception {
         final TestingJobMasterServiceLeadershipRunnerFactory jobManagerRunnerFactory =
                 new TestingJobMasterServiceLeadershipRunnerFactory(1);
         final TestingCleanupRunnerFactory cleanupRunnerFactory = new TestingCleanupRunnerFactory();
```

Comment:
Here below similar question about `dispatcherRunner` whether we need it or not...
And at the same time there is 
```java
final TestingJobManagerRunner testingJobManagerRunner =
                    jobManagerRunnerFactory.takeCreatedJobManagerRunner();
```
where some resources are opened without closing... may be put it in `try` resource

Reply:
> Here below similar question about dispatcherRunner whether we need it or not... (in case yes... may just add a comment)

It is also necessary for this test as `dispatcherRunner` will start `LeaderElectionService` and play the role of `LeaderContender`.

> where some resources are opened without closing... may be put it in try resource

Done.